pkg/platform/platform.go

```diff
@@ -27,8 +27,14 @@ type Platform interface {
 	Apply(*terraform.Executor) error
 	Destroy(*terraform.Executor) error
 	Initialize(*terraform.Executor) error
-	GetAssetDir() string
-	GetExpectedNodes() int
+	Meta() Meta
+}
+
+// Meta is a generic information format about the platform.
+type Meta struct {
```

Comment:
IMHO, not sure this mini-refactor is worth it. Maybe if you have a design doc in mind that I don't, is obvious, but for me it isn't why this is going in the right direction.

I'd either refactor and not do c&p and that stuff, or live with the current approach and duplication that it implies. This hybrid seems weird, and it's not clear that will need be lost after refactoring the platform code (because I don't know how it will look like)

Reply:
After the refactoring the `AssetDir` will most likely go away, `ExpectedNodes` may go away later and `Managed` will stay, as this is kind of information each platform should provide from their configuration. But I would rather not interfere with refactoring work at the moment. All those information will be available after the refactor anyway in platform code, so we can optimize it later to avoid complex data flow.

Reply:
I agree. But I don't follow, why are we changing the platform interface instead of continue to use it the way it is, then?

Reply:
I need to change it anyway, because I need platforms to expose new information for me, `Managed` field. IMO it's better and more future-proof to replace existing `Get*()` methods with `Meta()` and return required information via struct. This way, if new field needs to be added in the future, then you only need to add this field and the API does not change.

Reply:
Oh, right. You are replacing *all* the `Get*` that should have never existed, haven't noticed before. Cool!

I'd prefer this kind of changes to not be overlooked, and I'm unsure about the name. But I think it is fine for now, if a refactor is coming and you have more of that in mind :)